pkg/kubelet/status/manager.go

```diff
@@ -182,16 +195,42 @@ func (m *manager) TerminatePods(pods []*api.Pod) bool {
 				Terminated: &api.ContainerStateTerminated{},
 			}
 		}
-		select {
-		case m.podStatusChannel <- podStatusSyncRequest{pod, pod.Status}:
-		default:
+		newStatus := m.updateStatusInternal(pod, pod.Status)
+		if newStatus != nil {
+			select {
+			case m.podStatusChannel <- podStatusSyncRequest{pod.UID, *newStatus}:
```

Comment:
Ah I see. Sounds good. The "channel is full case" currently relies on our periodic housekeeping in kubelet, but would be good to simply fall back to the periodic sync.